Breaking name change in collections 2.1

Unity's collections package contains breaking name changes in 2.1 ([changelog](https://docs.unity3d.com/Packages/com.unity.collections@2.1/changelog/CHANGELOG.html)). GOAP is impacted because `NativeMultiHashMap` is renamed to `NativeParallelMultiHashMap`.

Currently, getting the package on 2022.2.9 leads to ` Library\PackageCache\com.crashkonijn.goap@ec36688846\Runtime\CrashKonijn.Goap.Resolver\GraphResolver.cs(15,17): error CS0246: The type or namespace name 'NativeMultiHashMap<,>' could not be found (are you missing a using directive or an assembly reference?)`

--- a/Package/Runtime/CrashKonijn.Goap.Resolver/GraphResolver.cs
+++ b/Package/Runtime/CrashKonijn.Goap.Resolver/GraphResolver.cs
@@ -12,7 +12,12 @@ namespace CrashKonijn.Goap.Resolver
     {
         private readonly List<Node> indexList;
         private readonly List<IAction> actionIndexList;
+        
+#if UNITY_COLLECTIONS_2_1
+        private NativeParallelMultiHashMap<int, int> map;
+#else
         private NativeMultiHashMap<int, int> map;
+#endif
         
         private GraphResolverJob job;
         private JobHandle handle;
@@ -25,8 +30,12 @@ namespace CrashKonijn.Goap.Resolver
             
             this.indexList = this.graph.AllNodes.ToList();
             this.actionIndexList = this.indexList.Select(x => x.Action).ToList();
-
+            
+#if UNITY_COLLECTIONS_2_1
+            var map = new NativeParallelMultiHashMap<int, int>(this.indexList.Count, Allocator.Persistent);
+#else
             var map = new NativeMultiHashMap<int, int>(this.indexList.Count, Allocator.Persistent);
+#endif
             
             for (var i = 0; i < this.indexList.Count; i++)
             {

--- a/Package/Runtime/CrashKonijn.Goap.Resolver/GraphResolverJob.cs
+++ b/Package/Runtime/CrashKonijn.Goap.Resolver/GraphResolverJob.cs
@@ -39,7 +39,11 @@ namespace CrashKonijn.Goap.Resolver
     public struct GraphResolverJob : IJob
     {
         // Graph specific
+#if UNITY_COLLECTIONS_2_1
+        [ReadOnly] public NativeParallelMultiHashMap<int, int> Connections;
+#else
         [ReadOnly] public NativeMultiHashMap<int, int> Connections;
+#endif
 
         // Resolve specific
         [ReadOnly] public RunData RunData;

--- a/Package/Runtime/CrashKonijn.Goap.Resolver/ResolveHandle.cs
+++ b/Package/Runtime/CrashKonijn.Goap.Resolver/ResolveHandle.cs
@@ -12,6 +12,20 @@ namespace CrashKonijn.Goap.Resolver
         private GraphResolverJob job;
         private RunData runData;
 
+#if UNITY_COLLECTIONS_2_1
+        public ResolveHandle(GraphResolver graphResolver, NativeParallelMultiHashMap<int, int> connections, RunData runData)
+        {
+            this.graphResolver = graphResolver;
+            this.job = new GraphResolverJob
+            {
+                Connections = connections,
+                RunData = runData,
+                Result = new NativeList<NodeData>(Allocator.TempJob)
+            };
+        
+            this.handle = this.job.Schedule();
+        }
+#else
         public ResolveHandle(GraphResolver graphResolver, NativeMultiHashMap<int, int> connections, RunData runData)
         {
             this.graphResolver = graphResolver;
@@ -24,6 +38,7 @@ namespace CrashKonijn.Goap.Resolver
         
             this.handle = this.job.Schedule();
         }
+#endif
 
         public IAction[] Complete()
         {

--- a/Package/Runtime/CrashKonijn.Goap.Resolver/com.crashkonijn.goap.resolver.asmdef
+++ b/Package/Runtime/CrashKonijn.Goap.Resolver/com.crashkonijn.goap.resolver.asmdef
@@ -13,6 +13,12 @@
     "precompiledReferences": [],
     "autoReferenced": true,
     "defineConstraints": [],
-    "versionDefines": [],
+    "versionDefines": [
+        {
+            "name": "com.unity.collections",
+            "expression": "2.1",
+            "define": "UNITY_COLLECTIONS_2_1"
+        }
+    ],
     "noEngineReferences": false
 }

